cpg-core/src/main/java/de/fraunhofer/aisec/cpg/passes/Pass.java

```diff
@@ -86,4 +118,50 @@ public void setLang(@Nullable LanguageFrontend lang) {
   public boolean supportsLanguageFrontend(LanguageFrontend lang) {
     return true;
   }
+
+  @NotNull
+  public Boolean isLastPass() {
+    try {
+      return this.getClass().isAnnotationPresent(ExecuteLast.class);
+    } catch (Exception e) {
+      return false;
+    }
+  }
+
+  @NotNull
+  public Boolean isFirstPass() {
+    try {
+      return this.getClass().isAnnotationPresent(FirstPass.class);
+    } catch (Exception e) {
+      return false;
+    }
+  }
+
+  public Set<Class<? extends Pass>> getSoftDependencies() {
+    return this.softDependencies;
+  }
+
+  public Set<Class<? extends Pass>> getHardDependencies() {
+    return hardDependencies;
+  }
+
+  /**
+   * Check whether the current language matches the language required by [RequiredLanguage]
+   *
+   * @return true, if the pass does not require a specific language frontend or if it matches the
+   *     [RequiredLanguage]
+   */
+  public boolean runWithCurrentFrontend() {
```

Comment:
`runWithCurrentFrontend` sounds like "execute this pass with current frontend" to me. Maybe `run*s*WithCurrentFrontend` or something that is more fitting to a boolean value?